src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const object of actions) {
+    for (const value of Object.values(object)) {
+      switch (value) {
+        case 'addProperties':
+          for (const [key1, value1] of Object.entries(object.extraData)) {
+            state[key1] = value1;
+          };
```

Comment:
I see that you really like loops, but would be more effectively to use `Object.assign` here